apps/gdalalg_vector_pipeline.cpp

```diff
@@ -566,8 +250,8 @@ std::string GDALVectorPipelineAlgorithm::GetUsageForCLI(
     {
         auto alg = GetStepAlg(name);
         assert(alg);
-        if (alg->CanBeFirstStep() && !alg->IsHidden() &&
-            name != GDALVectorReadAlgorithm::NAME)
+        if (alg->CanBeFirstStep() && !alg->CanBeMiddleStep() &&
```

Comment:
Given the default implemetation of CanBeMiddleStep() the check is not necessary, which makes me wonder if the default implementation is meant to be overridden by steps that can be an intermediate step and a first or last step at the same time. 

Thoughts?